src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -55,73 +38,42 @@ public abstract class Engine
 
         private readonly Func<Type, ILogger> loggerFactory;
 
-        private DataExtensionFactory factory;
-        private ExtensionRoot extensionRoot;
+        private readonly EngineCreateInfo createInfo;
+
+        private ILogger logger;
+        private List<ProcessingSourceExecutor> executors;
+        private ReadOnlyDataSourceSet workingDataSourceSet;
+        private DataSourceSet internalDataSourceSet;
+
+        private TableExtensionSelector tableExtensionSelector;
 
-        private IReadOnlyCollection<string> extensionDirectories;
         private bool isProcessed;
-        private IEnumerable<ErrorInfo> creationErrors;
 
         private IApplicationEnvironment applicationEnvironment;
-        private IAssemblyLoader loader;
         private bool isDisposed;
 
-        internal Engine(Func<Type, ILogger> loggerFactory)
+        internal Engine(EngineCreateInfo createInfo, DataSourceSet internalDataSourceSet)
```

Comment:
>internalDataSourceSet [](http://example.com/codeflow?start=67&length=21)

would you mind adding comment about internalDataSourceSet

Reply:
Sure